csaf-import/src/main/kotlin/io/github/csaf/sbom/retrieval/RetrievedProvider.kt

```diff
@@ -99,20 +128,24 @@ class RetrievedProvider(val json: Provider, val role: Role) {
             loader: CsafLoader = lazyLoader
         ): Result<RetrievedProvider> {
             val ctx = ValidationContext()
-            // Closure for providing HttpResponse to ValidationContext.
             // TODO: Only the last error will be available in result. We should do some logging.
             // First, we need to check if a .well-known URL exists.
             val wellKnownPath = "https://$domain/.well-known/csaf/provider-metadata.json"
             return loader
                 .fetchProvider(wellKnownPath, ctx)
-                .map { it.also { ctx.dataSource = ValidationContext.DataSource.WELL_KNOWN } }
+                .onSuccess { ctx.dataSource = ValidationContext.DataSource.WELL_KNOWN }
+                .mapCatching { p -> RetrievedProvider(p).also { it.validate(ctx) } }
```

Comment:
add this lambda to a variable and use it instead of having it three times?

Reply:
Not a big deal, but yes.